src/coreclr/vm/callhelpers.h

```diff
@@ -48,6 +48,12 @@ struct CallDescrData
 #else
     UINT64 returnValue;
 #endif
+    // True indicates that the new exception handling should rethrow managed exception
+    // using native EH / SEH when it reaches the CallDescrWorkerInternal.
+    // False means that the native code that called the CallDescrWorkerInternal will
+    // be skipped during EH. Please note that in this case destructors in that
+    // code won't be invoked, so extra care needs to be taken to avoid resource leaks.
```

Comment:
CallDescrWorkerInternal is used in like hundreds places. How are we making sure that it is fine to skip the destructors in all callers?

This looks like a very fragile solution.

Reply:
It is not skipping in any other callers than the `RuntimeMethodHandle::InvokeMethod`. The new flag in the CallDescrData is set to false only for that.

Reply:
Ah ok. Still, `RuntimeMethodHandle::InvokeMethod` is very complicated method with complicated macro expansions. It is very hard to review it to ensure that there are no important destructors in it. (I am actually surprised that you have not run into any - I guess that the test run will tell.)